expyfun/stimuli/_tracker.py

```diff
@@ -0,0 +1,837 @@
+    # Define all the public properties
+    # =========================================================================
+    @property
+    def alpha(self):
+        return self._alpha
+
+    @property
+    def chance(self):
+        return self._chance
+
+    @property
+    def max_trials(self):
+        return self._max_trials
+
+    @property
+    def stop_early(self):
+        return self._stop_early
+
+    @property
+    def p_val(self):
+        return self._p_val
+
+    @property
+    def min_p_val(self):
+        return self._min_p_val
+
+    @property
+    def max_p_val(self):
+        return self._max_p_val
+
+    @property
+    def n_trials(self):
+        return self._n_trials
+
+    @property
+    def n_wrong(self):
+        """The number of incorrect trials so far
+        """
+        return self._n_wrong
+
+    @property
+    def n_correct(self):
+        """The number of correct trials so far
+        """
+        return self._n_correct
+
+    @property
+    def pc(self):
+        """Proportion correct (0-1, NaN before any responses made)
+        """
+        return self._pc
+
+    @property
+    def responses(self):
+        """The response history
+        """
+        return self._responses
+
+    @property
+    def stopped(self):
+        """Is the tracker stopped
+        """
+        return self._stopped
+
+    @property
+    def success(self):
+        """Has the p-value reached significance
+        """
+        return self._p_val <= self._alpha
+
+    @property
+    def x_current(self):
+        """Included only for compatibility with TrackerDealer
+        """
+        return self._x_current
+
+    @property
+    def x(self):
+        """Included only for compatibility with TrackerDealer
+        """
+        return np.array([self._x_current for _ in range(self._n_trials)])
+
+    @property
+    def stop_rule(self):
+        return 'trials'
+
+
+# =============================================================================
+# Define a container for interleaving several tracks simultaneously
+# =============================================================================
+# TODO: Make it so you can add a list of values for each dimension (such as the
+# phase in a BMLD task) and have it return that
+
+# TODO: eventually, make a BaseTracker class so that TrackerDealer can make
+# sure it has the methods / properties it needs
+class TrackerDealer(object):
+    """Class for selecting and pacing independent simultaneous trackers
+
+    Parameters
+    ----------
+    shape : list-like
+        The dimensions of the tracker container.
+    tracker_class : class
+        The class for the tracker you want to run, e.g. ``TrackerUD``--not an
+        instance.
+    tracker_args : list-like
+        The arguments used to instantiate each of the trackers.
+    tracker_kwargs : dict
+        The keyword arguments used to instantiate each of the trackers.
+    slop : int
```

Comment:
Yes, and changed, but it does lead to this lovely little snippet: ``if lag_max > self._max_lag``.

Worth it, though, I think (as much as I would like to have a parameter called "slop").